tests/sentinel/tests/08-replica-priority.tcl

```diff
@@ -0,0 +1,66 @@
+source "../tests/includes/init-tests.tcl"
+
+test "Check acceptable replica-priority values" {
+    foreach_redis_id id {
+        if {$id == $master_id} continue
+
+        # ensure replica-announced accepts yes and no
+        catch {R $id CONFIG SET replica-announced no} e
+        if {$e ne "OK"} {
+            fail "Unable to set replica-announced to no"
+        }
+        catch {R $id CONFIG SET replica-announced yes} e
+        if {$e ne "OK"} {
+            fail "Unable to set replica-announced to yes"
+        }
+
+        # ensure a random value throw error
+        catch {R $id CONFIG SET replica-announced 321} e
+        if {$e eq "OK"} {
+            fail "Able to set replica-announced with something else than yes or no (321) whereas it should not be possible"
+        }
+        catch {R $id CONFIG SET replica-announced a3b2c1} e
+        if {$e eq "OK"} {
+            fail "Able to set replica-announced with something else than yes or no (a3b2c1) whereas it should not be possible"
+        }
+
+        # test only the first redis replica, no need to double test
+        break
+    }
+}
+
+proc 08_test_number_of_replicas {} {
+    uplevel 1 {
+        test "Check sentinel replies with replicas ignoring those with replica-announced=no" {
+            # count number of replicas
+            set n_replicas 0
+            foreach_redis_id id {
+                if {$id == $master_id} continue
+                if {[R $id CONFIG GET replica-announced] eq "replica-announced yes"} {
+                    incr n_replicas
+                }
+            }
```

Comment:
I suggest to be a bit more explicit here:
* Explicitly call `CONFIG SET replica-announced` from the same test and not rely on the left over state from the previous test (which could change)
* Use an explicit `n_replicas` rather than count it dynamically (e.g. consider there are TWO bugs: one that fails to provide the correct INFO output, and another in Sentinel's behavior).

Reply:
I rewrite this part, it should be better